### DISCUSSION TITLE
Misunderstandings regarding the integrate2D function have been resolved.

### DISCUSSION THREAD

**Discussion by Jkim9486** [silx-kit/pyFAI#2321] [Q&A]
Hello everyone,

I recently upgraded to pyFAI version 2024.09 from a much older release, and I’ve encountered a change in ai.integrate2d that I’d like some clarity on.

In the older version, I used ai.integrate2d to separate isotropic (iso) and anisotropic (aniso) signals in my images. Specifically, results[0] contained the iso signal, and results[2] provided the aniso signal. However, in the current version, the output appears to only contain azimuthal information, and I am no longer able to directly access the isotropic and anisotropic signals as I could before.

Since the purpose of integrate2d includes distinguishing between the isotropic and anisotropic signals, it’s essential for my workflow to have explicit access to them. I’m curious if there’s an updated approach to retrieve both signals separately or if this functionality has been altered in the latest release.

Any insights or guidance from those familiar with recent changes in integrate2d would be greatly appreciated!

Thanks in advance!

**Comment by kif** [comment]
By any chance, is this what you are looking for:
http://www.silx.org/doc/pyFAI/latest/api/pyFAI.html#pyFAI.azimuthalIntegrator.AzimuthalIntegrator.separate
